manager/src/main/java/com/redhat/service/bridge/manager/BridgeWorker.java

```diff
@@ -0,0 +1,208 @@
+            LOGGER.warn("The status of the entity is not null. It will be overwritten.");
+        }
+        bridge.setStatus(ManagedEntityStatus.ACCEPTED);
+        bridge.setDesiredStatus(ManagedEntityStatus.PREPARING);
+
+        // Persist and fire
+        transact(bridge);
+
+        PreparingWorker preparingWorker = new PreparingWorker();
+        preparingWorker.setEntityId(bridge.getId());
+        preparingWorker.setStatus(null); // where the resource is
+        preparingWorker.setDesiredStatus(BridgeWorkerStatus.TOPIC_CREATED.toString()); // where you want to go. i.e. step 1
+        preparingWorker.setWorkerId(workerIdProvider.getWorkerId()); // The workerId is set immediately
+        preparingWorker.setSubmittedAt(ZonedDateTime.now(ZoneOffset.UTC));
+        preparingWorker.setModifiedAt(ZonedDateTime.now(ZoneOffset.UTC));
+        preparingWorker.setType("BRIDGE");
+        transact(preparingWorker);
+
+        routePreparingWorker(preparingWorker, bridge);
+    }
+
+    @Override
+    public void deprovision(Bridge bridge) {
+        LOGGER.info("Deprovision entity " + bridge.getId());
+
+        bridge.setStatus(ManagedEntityStatus.DEPROVISION_ACCEPTED);
+        bridge.setDesiredStatus(ManagedEntityStatus.DEPROVISION);
+
+        // Persist and fire
+        transact(bridge);
+
+        PreparingWorker preparingWorker = new PreparingWorker();
+        preparingWorker.setEntityId(bridge.getId());
+        preparingWorker.setStatus(null); // where the resource is
+        preparingWorker.setDesiredStatus(BridgeWorkerStatus.TOPIC_DELETED.toString()); // where you want to go
+        preparingWorker.setWorkerId(workerIdProvider.getWorkerId());
+        preparingWorker.setSubmittedAt(ZonedDateTime.now(ZoneOffset.UTC));
+        preparingWorker.setModifiedAt(ZonedDateTime.now(ZoneOffset.UTC));
+        transact(preparingWorker);
+
+        routePreparingWorker(preparingWorker, bridge);
+    }
+
+    @Transactional
+    @ConsumeEvent(value = CREATE_KAFKA_TOPIC, blocking = true)
+    public void step1_createKafkaTopic(Bridge bridge) {
+        try {
+            rhoasService.createTopicAndGrantAccessFor(internalKafkaConfigurationProvider.getBridgeTopicName(bridge), RhoasTopicAccessType.CONSUMER_AND_PRODUCER);
+        } catch (RuntimeException e) {
+            LOGGER.error("Failed to create the topic. The retry will be performed by the reschedule loop.", e);
+            updatePreparingWorker(bridge.getId(), BridgeWorkerStatus.FAILURE);
+            return;
+        }
+
+        PreparingWorker preparingWorker = preparingWorkerDAO.findByEntityId(bridge.getId()).get(0); // there should be no parallel workers for this resource.
+        preparingWorkerDAO.deleteById(preparingWorker.getId());
+
+        // The bridge is ready to be provisioned by the shard.
+        bridge.setStatus(ManagedEntityStatus.PREPARING);
+        bridge.setDesiredStatus(ManagedEntityStatus.PROVISIONING);
+        transact(bridge);
+    }
+
+    @Transactional
```

Comment:
@r00ta We need to be careful here. The transaction boundary on this method involves the call to the remote service. Part of the reason to introduce the async model is to avoid calling a remote service within the boundaries of a DB transaction.